docs/cli-plugins.md

```diff
@@ -340,6 +340,36 @@ config dir, if needed, using something like
 you'll need to decide how to handle differences in preferences across
 different versions of your plugin.
 
+#### $EXEC_ARGS_FILE
+
+This is the path of a file that contains the args that would be passed to an
+executable built by bazel as a result of a `bazel run` command. Specifically,
+these are any positional arguments remaining after canonicalization of any
+options that take arguments into "--option=value" options, excepting the `run`
+subcommand itself and the build target. These are generally the arguments
+following a `--` in the argument list passed to bazel, if any.
```

Comment:
nit: I think it's a bit confusing to mention canonicalization here, since it doesn't change the args after `--`. wdyt about removing that sentence?

```suggestion
executable built by bazel as a result of a `bazel run` command. These are the arguments
following a `--` in the argument list passed to bazel, if any.
```

Reply:
So, technically, all `--` does is mark the following arguments as positional args (as opposed to options), and what bazel actually does is pass any positional args following the `run` subcommand and a single target (the first two positional arguments) which do not have a corresponding option to the resulting executable. People normally use `--` because some of the args they want to pass will begin with `-` or `--` (because they want the resulting executable to interpret them as options), and without the `--` bazel would try to interpret them as options instead of parsing them as positional args and forwarding them to the resulting executable.

It is technically valid, for instance, to call `bazel run -- //server --help`, and it will print the help text for the server. On testing and code inspection, though, it seems the buildbuddy cli is actually just partitioning on the `--` to separate these two sets of arguments (for the purposes of the files with lists of arguments to be edited), which is not really correct.

I have added a test to demonstrate this behavior, and a `TODO` about fixing it.

Reply:
Note that bazel interprets the next argument after an option that takes a positional argument as a value for that option, even if that argument is `--`, so there is no need to worry about positional arguments *after* a `--` potentially being values for a bazel option.

Reply:
Ah, TIL. SG to fix this later :)